transformation/plugins/org.jboss.tools.fuse.transformation.editor/src/main/java/org/jboss/tools/fuse/transformation/editor/wizards/NewTransformationWizard.java

```diff
@@ -434,14 +432,16 @@ private String generateModel(String filePath,
             return getJavaQualifiedName(filePath, monitor);
         }
         case JSON: {
+            String className = classNameBuilder.toString().replace("_", "");
```

Comment:
Well, that didn't turn out well at all.  it turns out we need the generated class name for subsequent use by the tooling, but the generator not only doesn't return that directly, but doesn't provide any way that I can discern to distinguish the actual top-level class we want to map from the other top-level classes it generates, at least not with some significant additional processing and code.  So, for the time being I'm going to leave the code as is with the underscore being dealt with in the wizard.  The worst impact of this is that until I figure out a way to determine the class to be mapped from the generator, I can't effectively produce a good regression test for this either.